.github/actions/setup_software/action.yaml

```diff
@@ -0,0 +1,72 @@
+name: 'Install software'
+description: 'Install software packages'
+inputs:
+  cache-name:  # id of input
+    description: 'Name to generate the full cache name'
+    required: true
+    default: 'cache-software-basis'
+  filesForHash:
+    description: 'List of files to be used for cache hash generation'
+    required: true
+    # define default. IMPORTANT: This is already a multiline string
+    # so DO NOT use any quotes!
+    default: |
+      **/requirements*.txt
+      **/sandboxes/cf_*.sh
+  run_additional:
+    description: "path to additional shell script to source"
+  cache_paths:
+    description: "paths to include in the caching"
+    required: true
+    # define default. IMPORTANT: This is already a multiline string
+    # so DO NOT use any quotes!
+    default: |
+      data/software/conda
+      data/software/venvs/cf_*
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout ⬇️
+      uses: actions/checkout@master
+      with:
+        persist-credentials: false
+        submodules: recursive
+        clean: false
+
+    - name: Setup python 🐍
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.10"
+
+    - name: Install dependencies ☕️
+      id: install_cache
+      uses: actions/cache@v3
+      env:
+        cache-name: ${{ inputs.cache-name }}
+      with:
+        path: |
+          ${{ inputs.cache_paths }}
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles(inputs.filesForHash) }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
```

Comment:
Are these needed? Seems like there should be no need to specify `restore-keys` here since one always passes the `cache-name` in the relevant task and the key should correspond exactly to the `key` entry as defined above.